File: src/main/java/io/github/tanguygab/conditionalactions/actions/types/messages/BroadcastActionBarAction.java
package io.github.tanguygab.conditionalactions.actions.types.messages;

import io.github.tanguygab.conditionalactions.actions.types.Action;
import net.md_5.bungee.api.ChatMessageType;
import net.md_5.bungee.api.chat.BaseComponent;
import net.md_5.bungee.api.chat.TextComponent;
import org.bukkit.OfflinePlayer;

public class BroadcastActionBarAction extends Action {

    public BroadcastActionBarAction() {
        super("^(?i)((broadcast|bc)-actionbar):( )?");
    }

    @Override
    public String getSuggestion() {
        return "broadcast-actionbar: <message>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        match = parsePlaceholders(player, match, true);
        BaseComponent[] components = TextComponent.fromLegacyText(match);
        getPlugin().getServer().getOnlinePlayers().forEach(p->p.spigot().sendMessage(ChatMessageType.ACTION_BAR,components));
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/messages/BroadcastActionBarAction.kt
package io.github.tanguygab.conditionalactions.actions.types.messages

import io.github.tanguygab.conditionalactions.actions.Action
import org.bukkit.OfflinePlayer

class BroadcastActionBarAction : Action("^(?i)((broadcast|bc)-actionbar):( )?".toRegex()) {
    override fun getSuggestion() = "broadcast-actionbar: <message>"

    override fun execute(player: OfflinePlayer?, match: String) {
        val match = parsePlaceholders(player, match)
        plugin.server.sendActionBar(mm.deserialize(match))
    }
}
